overlord/ifacestate/helpers.go

```diff
@@ -240,6 +243,11 @@ func (m *InterfaceManager) regenerateAllSecurityProfiles(tm timings.Measurer) er
 		return precompOpts[instanceName]
 	}
 
+	if unlockState {
+		m.state.Unlock()
+		defer m.state.Lock()
+	}
```

Comment:
I wonder whether we should put this into a closure surrounding the for loop instead

Reply:
done